ql/indexes/iborindex.cpp

```diff
@@ -82,6 +82,84 @@ namespace QuantLib {
    : IborIndex(familyName, 1*Days, settlementDays, curr,
                fixCal, Following, false, dc, h) {}
 
+    void OvernightIndex::addFixing(const Date& fixingDate, Real fixing, bool forceOverwrite) {
+        calculated_ = false;
+        Index::addFixing(fixingDate, fixing, forceOverwrite);
+    }
+
+    void OvernightIndex::addFixings(const TimeSeries<Real>& t, bool forceOverwrite) {
+        calculated_ = false;
+        Index::addFixings(t, forceOverwrite);
+    }
+
+    Rate OvernightIndex::compoundedFixings(const Date& fromFixingDate, const Date& toFixingDate) {
+        calculate();
+        auto yearFraction = dayCounter_.yearFraction(fromFixingDate, toFixingDate);
+        auto compIndexEnd = compoundIndex_[toFixingDate];
+        auto compIndexStart = compoundIndex_[fromFixingDate];
+
+        if (compIndexStart == Null<Real>() || compIndexEnd == Null<Real>())
+            return Null<Real>();
+
+        return (compoundIndex_[toFixingDate] / compoundIndex_[fromFixingDate] - 1) / yearFraction;  
+    }
+
+    Real OvernightIndex::compoundedFactor(const Date& fromFixingDate, const Date& toFixingDate) {
+        calculate();
+        auto compIndexEnd = compoundIndex_[toFixingDate];
+        auto compIndexStart = compoundIndex_[fromFixingDate];
+
+        if (compIndexStart == Null<Real>() || compIndexEnd == Null<Real>())
+            return Null<Real>();
+
+        return compIndexEnd / compIndexStart;
+    }
+ 
+    void OvernightIndex::performCalculations() const {
```

Comment:
I'm not sure LazyObject is the right mechanism for this.  This calculation will be executed again and again every time the index (or the pricer, if you move the calculation there) receives a notification; for instance, when the forecast curve changes, even though it has no effect on the stored past prices on which this calculation depends.

Reply:
Oh, I see, because you've removed the part in `update` where the lazy object sets `calculated_` to false.  Hmm, yes, it would work, but I'm not sure if I'm comfortable with changing the way the lazy object works.  Also, you might add a fixing through a different instance of the index, and this instance would still have `calculated_` set to false and ignore the new fixing.